lib/deep_merge.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+function deepMerge(target, source) {
+  target = Object.assign({}, target);
+  source = Object.assign({}, source);
+
+  // Iterate through `source` properties and if an `Object` set property to merge of `target` and `source` properties
+  for (const key of Object.keys(source)) {
+    if (source[key] instanceof Object) Object.assign(source[key], deepMerge(target[key], source[key]));
+  }
+
+  // Join `target` and modified `source`
+  Object.assign(target || {}, source);
```

Comment:
I don't think `|| {}` is necessary since `target` is never a null; even if `target` is null, `|| {}` doesn't prevent error.

Reply:
It is used to handle if `target` is `undefined`, so `deepMerge(obj)` will return a shallow copy of `obj`.

Reply:
When `deepMerge(obj)`, `obj` is passed as `target` parameter, so when `obj` is `undefined`, `deepMerge(obj)` it will return `undefined`.

Anyhow, since L4 needs to be removed, `|| {}` is indeed necessary to avoid error when `target` is `undefined`.